documentation/learningPath/testing.md

```diff
@@ -2,3 +2,86 @@
+* The [AsyncWaitScenario](../../src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/AsyncWaitScenario.cs) is sufficient for most tests.
+
+## Native/Profiler Tests
+
+- [Microsoft.Diagnostics.Monitoring.Profiler](../../src/tests/Microsoft.Diagnostics.Monitoring.Profiler/)
+- [Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp](../../src/tests/Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp/)
+
+This test assembly provides a test to make sure the dotnet-monitor profiler can load into a target app.
+
+## Schema Generation
+
+- [Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests](../../src/tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests/)
+- [Microsoft.Diagnostics.Monitoring.ConfigurationSchema](../../src/tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/)
+- [Microsoft.Diagnostics.Monitoring.Options](../../src/Microsoft.Diagnostics.Monitoring.Options)
+
+Dotnet-monitor generates [schema.json](../../documentation/schema.json) using unit tests. If dotnet-monitor configuration changes, the schema.json file needs to be updated.
+Note that it is possible to compile option classes directly into the `ConfigurationSchema` project. This may be necessary in order to attribute properties appropriately for schema generation. See [Microsoft.Diagnostics.Monitoring.ConfigurationSchema.csproj](../../src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/Microsoft.Diagnostics.Monitoring.ConfigurationSchema.csproj).
+
+## OpenAPI generation
+
+- [Microsoft.Diagnostics.Monitoring.OpenApiGen.UnitTests](../../src/tests/Microsoft.Diagnostics.Monitoring.OpenApiGen.UnitTests/)
+- [Microsoft.Diagnostics.Monitoring.OpenApiGen](../../src/tests/Microsoft.Diagnostics.Monitoring.OpenApiGen/)
+
+These assemblies and tests are used to generate the [OpenAPI spec](../../documentation/openapi.json) for the dotnet-monitor API. Changes to the dotnet-monitor api surface require updating openapi.json.
+
+## Startup hooks / hosting startup
+
+- [Microsoft.Diagnostics.Monitoring.Tool.TestStartupHook](../../src/tests/Microsoft.Diagnostics.Monitoring.Tool.TestStartupHook/)
+
+This assembly is injected into a dotnet-monitor runner (using `DOTNET_STARTUP_HOOKS`) to facilitate Assembly resolution during test runs.
+
+- [Microsoft.Diagnostics.Monitoring.Tool.TestHostingStartup](../../src/tests/Microsoft.Diagnostics.Monitoring.Tool.TestHostingStartup/)
+
+Uses `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` to inject a service into dotnet-monitor during test time. This allows tests to locate files that are not normally part of the test deployment,
+such as the native profiler.
+
+- [Microsoft.Diagnostics.Monitoring.StartupHook.UnitTests](../../src/tests/Microsoft.Diagnostics.Monitoring.StartupHook.UnitTests/)
+
+Unit tests around features that are injected via `DOTNET_STARTUP_HOOKS` into the target application. This currently includes the Exceptions History feature.
```

Comment:
nit: It might be helpful for "Exceptions History feature" to link to something for anyone who wants to know more about what that is.

Reply:
I don't think we have docs for that yet.